### PR TITLE
[image-picker][ios] Fix unexpected behaviour of quality option on iOS

### PR DESCRIPTION
**Pull request by behenate**
# Why
On iOS when using ImagePicker with `.bmp` files the file would be converted to `.png` if any quality is selected, even though  the option should be ignored. Therefore setting quality to 0.2 (default value) would yield different results  than for not setting it at all  https://github.com/expo/expo/issues/20299

# How

Updated the `ImagePickerModule` to ignore the quality option with `.bmp` files.

When opening a `.bmp` image with `allowsEditing` set to `true` file has to be converted to `.png`, now the `fileName` is also updated to have `.png` file extension. Docs have been updated to make it more clear when the conversion happens.

# Test Plan

Tested on a physical device with opening images from gallery and the camera.

### CODE REVIEW

File: packages/expo-image-picker/ios/MediaHandler.swift [hunk 1]
```diff
@@ -338,7 +342,7 @@ private struct ImageUtils {
       return (data, ".png")
 
     case .some(let s) where s.contains("ext=BMP"):
-      if options.allowsEditing || options.quality != nil {
+      if options.allowsEditing {
```
Comment (barthap) [inline packages/expo-image-picker/ios/MediaHandler.swift:345]: Good catch! The `quality` should be ignored. This looks correct now

File: packages/expo-image-picker/src/ImagePicker.types.ts [hunk 2]
```diff
@@ -215,7 +215,8 @@ export type ImagePickerAsset = {
   /**
    * Preferred filename to use when saving this item. This might be `null` when the name is unavailable
    * or user gave limited permission to access the media library.
-   *
+   * > `fileName` will be the same as the original filename, except when opening `.bmp` files with `allowsEditing` set to `true`.
+   *  In this case the image will be converted to `.png` and the `fileName` will be updated accordingly.
```
Comment (barthap) [inline packages/expo-image-picker/src/ImagePicker.types.ts:219]:
```suggestion
   *
```

IMO this is redundant. The doc comment for the `allowsEditing` option is enough.

**Review by barthap — CHANGES_REQUESTED** [review]
Thank you for the fix! The previous behaviour was incorrect indeed.
The actual fix code (switch cases for BMP) looks good! 🎉 

But please try doing the filename replacement the Swift way. Personally, I'd create a helper function, which looks sth like this (written by hand, probably wouldn't compile):
```swift
func replaceFileExtension(fileName: String?, targetExtension: String) -> String? {
  guard let fileName = fileName else { return nil }
  if !fileName.lowercased.contains(targetExtension.lowercased()) {
    return fileName.deletingPathExtension + targetExtension
  }
  return fileName
}
```
Also, please use Swift-specific methods for that, as linter suggests.

If you need more help with this, feel free to ping me

Edit: One more thing. You need to run `yarn build` from inside the `packages/expo-image-picker` directory and commit the generated `build/*` files for CI to pass

File: packages/expo-image-picker/CHANGELOG.md [hunk 3]
```diff
@@ -7,7 +7,7 @@
 ### 🎉 New features
 
 ### 🐛 Bug fixes
-
+- Fix images unexpectedly being converted to `.png` when opening `.bmp` files and selecting any quality in `ImagePickerOptions`. ([#21361](https://github.com/expo/expo/pull/21361) by [@behenate](https://github.com/behenate))
```
Comment (lukmccall) [inline packages/expo-image-picker/CHANGELOG.md:10]:
```suggestion

- Fix images unexpectedly being converted to `.png` when opening `.bmp` files and selecting any quality in `ImagePickerOptions`. ([#21361](https://github.com/expo/expo/pull/21361) by [@behenate](https://github.com/behenate))

```

File: packages/expo-image-picker/ios/MediaHandler.swift [hunk 4]
```diff
@@ -251,6 +255,25 @@ internal struct MediaHandler {
 
   // MARK: - utils
 
+  private func replaceFileExtension(fileName: String?, targetExtension: String) -> String? {
```
Comment (lukmccall) [inline packages/expo-image-picker/ios/MediaHandler.swift:258]:
```suggestion
  private func replaceFileExtension(fileName: String, targetExtension: String) -> String? {
```
I think that here we are sure that the `fineName` can't be null.

Comment (behenate) [reply packages/expo-image-picker/ios/MediaHandler.swift]: You are right the String? and guard let were unnecessary

File: packages/expo-image-picker/src/ImagePicker.types.ts [hunk 5]
```diff
@@ -215,7 +215,6 @@ export type ImagePickerAsset = {
   /**
    * Preferred filename to use when saving this item. This might be `null` when the name is unavailable
    * or user gave limited permission to access the media library.
-   *
```
Comment (barthap) [inline packages/expo-image-picker/src/ImagePicker.types.ts:218]: A leftover?

**Review by barthap — APPROVED** [review]
LGTM

**Review by lukmccall — APPROVED** [review]

**Pull request merged**